### PR TITLE
processing: add option to select directory to recurse for files

### PR DESCRIPTION
**Pull request by jef-n**
fixes regressions with a number of tools that used to have this - eg. gdalbuildvrt, gdaladdo

### CODE REVIEW

File: python/plugins/processing/algs/gdal/GdalUtils.py [hunk 1]
```diff
@@ -405,15 +405,18 @@ def parseCreationOptions(value):
...
-            if executing:
-                layers = []
-                for l in alg.parameterAsLayerList(parameters, parameter_name, context):
-                    if quote:
-                        layers.append('"' + l.source() + '"')
-                    else:
-                        layers.append(l.source())
+
+        layers = []
+        for l in alg.parameterAsLayerList(parameters, parameter_name, context):
```
Comment (nyalldawson) [inline python/plugins/processing/algs/gdal/GdalUtils.py]: This is an expensive call - it should NOT be made when not executing.

**Comment by nyalldawson** [comment]
Please give time for others to review changes like this -- recent prs and reversions in batch dialog have shown have fragile this area of code is, everything here NEEDs to be reviewed by @alexbruy or we risk further regressions.

**Comment by alexbruy** [comment]
Looks good to me

**Comment by jef-n** [comment]
> Please give time for others to review changes like this -- recent prs and reversions in batch dialog have shown have fragile this area of code is, everything here NEEDs to be reviewed by @alexbruy or we risk further regressions.

Fixed in 19933e0f

**Comment by nyalldawson** [comment]
Thanks @jef-n!

**Pull request merged**